### PR TITLE
fix(tree-select): localization for hiddenItemsText (#DS-2680)

### PR DESCRIPTION
**Pull request by NikGurev**
## Summary
Added localization for "show more" text

## List of notable changes
**BREAKING CHANGES**: previous solution with `hiddenItemsTextFormatter` function will be broken. To support new functionality, just remove it.

To support custom data for "show more" text provide `hiddenItemsText` and `hiddenItemsTextFormatter` simultaneously

### CODE REVIEW

File: packages/components/tree-select/tree-select.component.ts [hunk 1]
```diff
@@ -472,10 +472,13 @@ export class KbqTreeSelect
         @Optional() parentForm: NgForm,
         @Optional() parentFormGroup: FormGroupDirective,
         @Optional() private readonly parentFormField: KbqFormField,
-        @Optional() @Self() ngControl: NgControl
+        @Optional() @Self() ngControl: NgControl,
+        @Optional() @Inject(KBQ_LOCALE_SERVICE) private localeService?: KbqLocaleService
     ) {
         super(elementRef, defaultErrorStateMatcher, parentForm, parentFormGroup, ngControl);
 
+        this.localeService?.changes.subscribe(this.updateLocaleParams);
```
Comment (artembelik) [inline packages/components/tree-select/tree-select.component.ts]: unsubscribe или [takeUntilDestroyed](https://angular.dev/api/core/rxjs-interop/takeUntilDestroyed)

File: packages/components-dev/tree-select/template.html [hunk 2]
```diff
@@ -415,6 +410,7 @@ <h2 class="kbq-subheading">Default footer with icon button</h2>
                     placeholder="this is placeholder"
                     [disabled]="disabledState"
                     [formControl]="control"
+                    [hiddenItemsText]="'еще'"
```
Comment (artembelik) [inline packages/components-dev/tree-select/template.html:413]: а почему не дефолтное значение (которое в компоненте указал)?

Comment (NikGurev) [reply packages/components-dev/tree-select/template.html:413]: Это для кастомизации:
> To support custom data for "show more" text provide hiddenItemsText and hiddenItemsTextFormatter simultaneously

File: packages/components/tree-select/tree-select.component.ts (hunk 1 above)
Comment (NikGurev) [reply packages/components/tree-select/tree-select.component.ts]: добавил `this.localeService?.changes.pipe(takeUntil(this.destroy)).subscribe(this.updateLocaleParams);`

**Review by artembelik — APPROVED** [review]

**Review by lskramarov — APPROVED** [review]

**Pull request merged**